frb_codegen/src/generator/rust/ty_primitive_list.rs

```diff
@@ -8,18 +8,29 @@ type_rust_generator_struct!(TypePrimitiveListGenerator, IrTypePrimitiveList);
 
 impl TypeRustGeneratorTrait for TypePrimitiveListGenerator<'_> {
     fn wire2api_body(&self) -> Option<String> {
-        Some(
-            "unsafe {
-            let wrap = support::box_from_leak_ptr(self);
-            support::vec_from_leak_ptr(wrap.ptr, wrap.len)
-        }"
+        Some(match &self.ir {
+            IrTypePrimitiveList::PrimitiveList { .. } => "unsafe {
+                    let wrap = support::box_from_leak_ptr(self);
+                    support::vec_from_leak_ptr(wrap.ptr, wrap.len)
+                }"
             .into(),
-        )
+            IrTypePrimitiveList::PrimitiveArray { len, .. } => format!(
```

Comment:
Is it possible we *delegate* a PrimitiveArray using a PritmitiveList, such that we even do not need to deal with any unsafe operations!

When talking about "delegate" - for example, when implementing a `String`, we delegate it to a `Vec<u8>`. See ty_delegate.rs for details.

Reply:
I used `delegate` when I first implemented it but got a response about whether it could be a separate module from list, the new implementation is with list and has some differences

Reply:
> but got a response about whether it could be a separate module from list

Did I do that? Could you please point out my comments (I am so forgetful haha since I am so busy recently)

I will recheck why I said this should not use a delegate.

Reply:
> In addition, I am not sure whether adding IrTypeDelegate::ArrayGeneral is the best approach. We already have existing support for arrays indeed. https://github.com/fzyzcjy/flutter_rust_bridge/pull/620#issuecomment-1207290477

Do you mean this one?

I see. Then yes, we should make array have only one IR, not two. But that does not conflict with delegating: What about this - we put array IR the same as current PR, but use delegating when *generating code*?

Reply:
This may need a bit of refactor. Namely:

* Maybe we need to enable *every* type to be a delegate, instead of "only types in IRDelegate can be delegate".
* After that, we can simply rename IRDelegate to something else, say, IRMisc, since they are no longer the only delegates.